sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/csv/UnivocityParserSuite.scala

```diff
@@ -267,4 +269,63 @@ class UnivocityParserSuite extends SparkFunSuite with SQLHelper {
     assert(convertedValue.isInstanceOf[UTF8String])
     assert(convertedValue == expected)
   }
+
+  test("skipping rows using pushdown filters") {
+    def check(
+        input: String = "1,a",
+        dataSchema: String = "i INTEGER, s STRING",
+        requiredSchema: String = "i INTEGER",
+        filters: Seq[Filter],
+        expected: Seq[InternalRow]): Unit = {
+      def getSchema(str: String): StructType = str match {
```

Comment:
@MaxGekk, it's a nit but I wouldn't add this nested function. Nested function is discouraged in general as it makes difficult to read. How about just simply doing `requiredSchema = StructType.fromDDL("i INTEGER, s STRING")` or `requiredSchema = new StructType()` in the caller sides? I think such one or two line duplications are fine.